### ISSUE TITLE
coreclr + vsdbg failure to get stack trace from remote Kubernetes pod

### ISSUE THREAD

**Issue by arminbashizade** [dotnet/runtime#101310]
I'm trying to attach a debugger on VS Code to a dotnet process running in a Kubernetes pod on a remote private cluster. I've followed the instructions from these guides:

- https://github.com/dotnet/vscode-csharp/wiki/Attaching-to-remote-processes
- https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/master/debug-on-kubernetes.md

I've installed vsdbg on the pod and the dotnet app is built in Debug mode. I can successfully attach the debugger, and it pauses the threads when it hits a breakpoint, however when it pauses it doesn't highlight the line of code where it stopped, doesn't show variable values or stacktrace. I've turned on diagnostic logs in launch.json and I see these errors in Debug Console when it hits the breakpoint (the most human-readable part being "Error processing 'stackTrace' request. Unknown Error: 0x80131124"):

```
[Debug Runtime Event] DBI Callback -- Breakpoint(AppDomain:00007F0A00000E48, Thread:00007F0A000610A8, Breakpoint:00007F0A01D826E8)
<- (E) {"seq":1605,"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/Attach","data":{"VS.Diagnostics.Debugger.vsdbg.Attach.DebuggerStartTime":976,"VS.Diagnostics.Debugger.vsdbg.SourceFileMappings":1,"VS.Diagnostics.Debugger.vsdbg.VisualizerFileUsed":false,"VS.Diagnostics.Debugger.vsdbg.TargetType":"Live","VS.Diagnostics.Debugger.vsdbg.TargetCLRVersion":"8.0.424.16909","VS.Diagnostics.Debugger.vsdbg.AdapterId":"coreclr","VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"12","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"debian","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.Version":"17.10.20209.7 commit:3e71548134e4af43d713feba19f1e9331b2d4a5e"}}}
[DebugEngine API] RETURN ERROR: IDkmThreadContextOperation.GetContext (hr = 0x80131124, ThreadId=2262, TickCount=156106459)
[DebugEngine API] RETURN ERROR: IDkmThreadContextOperation.GetContext (hr = 0x80131124, ThreadId=2262, TickCount=156106463)
[DebugEngine API] RETURN ERROR: IDkmStackWalkDispatcherService.GetCurrentRegisters (hr = 0x80131124, ThreadId=2262, TickCount=156106463)
[DebugEngine API] RETURN ERROR: IDkmMergedMonitorStackWalk164.RuntimeWalkNextFramesAndCheckCache164 (hr = 0x80131124, ThreadId=2262, TickCount=156106463)
[DebugEngine API] RETURN ERROR: IDkmStackProvider.GetTopStackFrame (hr = 0x80131124, ThreadId=2262, TickCount=156106463)
<- (E) {"seq":1611,"type":"event","event":"stopped","body":{"reason":"breakpoint","threadId":2155,"allThreadsStopped":true,"hitBreakpointIds":[1]}}
-> (C) {"command":"threads","type":"request","seq":11}
<- (R) {"seq":1612,"type":"response","request_seq":11,"success":true,"command":"threads","body":{"threads":[{"id":2155,"name":".NET TP Worker"},{"id":24,"name":"<No Name>"},{"id":2066,"name":".NET TP Worker"},{"id":30,"name":"Dynatrace.OneAgent.NET.ThreadPoolMetrics"},{"id":32,"name":".NET Timer"},{"id":2216,"name":".NET TP Worker"},{"id":40,"name":"<No Name>"},{"id":25,"name":"<No Name>"},{"id":41,"name":"Kestrel Timer"},{"id":29,"name":".NET Counter Poller"},{"id":2152,"name":".NET TP Worker"},{"id":2067,"name":".NET TP Worker"},{"id":39,"name":"<No Name>"},{"id":36,"name":".NET Sockets"},{"id":31,"name":".NET File Watcher"},{"id":1,"name":"<No Name>"},{"id":1939,"name":".NET TP Worker"},{"id":16,"name":"<No Name>"},{"id":37,"name":"Console logger queue processing thread"},{"id":34,"name":".NET TP Gate"},{"id":2263,"name":".NET TP Worker"},{"id":2264,"name":".NET TP Worker"},{"id":2265,"name":"<No Name>"}]}}
-> (C) {"command":"stackTrace","arguments":{"threadId":2155,"startFrame":0,"levels":20},"type":"request","seq":12}
[DebugEngine API] RETURN ERROR: IDkmThreadContextOperation.GetContext (hr = 0x80131124, ThreadId=2267, TickCount=156106583)
[DebugEngine API] RETURN ERROR: IDkmThreadContextOperation.GetContext (hr = 0x80131124, ThreadId=2267, TickCount=156106587)
[DebugEngine API] RETURN ERROR: IDkmStackWalkDispatcherService.GetCurrentRegisters (hr = 0x80131124, ThreadId=2267, TickCount=156106587)
[DebugEngine API] RETURN ERROR: IDkmMergedMonitorStackWalk164.RuntimeWalkNextFramesAndCheckCache164 (hr = 0x80131124, ThreadId=2267, TickCount=156106587)
[DebugEngine API] FIRE COMPLETION ROUTINE (ERROR): IDkmStackProvider.GetNextFrames from class StackProvider::CStackProviderObj (hr = 0x80131124, ThreadId=2267, TickCount=156106591)
<- (E) {"seq":1618,"type":"event","event":"output","body":{"category":"telemetry","output":"VS/Diagnostics/Debugger/vsdbg/CommandFailed","data":{"VS.Diagnostics.Debugger.vsdbg.ErrorCode":-2146234076,"VS.Diagnostics.Debugger.vsdbg.Command":"stackTrace","VS.Diagnostics.Debugger.vsdbg.TargetCLRVersion":"8.0.424.16909","VS.Diagnostics.Debugger.vsdbg.AdapterId":"coreclr","VS.Diagnostics.Debugger.vsdbg.Distribution.Version":"12","VS.Diagnostics.Debugger.vsdbg.Distribution.Name":"debian","VS.Diagnostics.Debugger.vsdbg.OSFamily":"Linux","VS.Diagnostics.Debugger.vsdbg.Version":"17.10.20209.7 commit:3e71548134e4af43d713feba19f1e9331b2d4a5e"}}}
<- (R) {"seq":1619,"type":"response","request_seq":12,"success":false,"command":"stackTrace","message":"Error processing 'stackTrace' request. Unknown Error: 0x80131124"}
```

I've tried running the same docker image in a local cluster with Rancher Desktop and it works as expected, stops at the breakpoint and shows variable values, etc.

**My question is:** Is there anything I can do to get more information about the error? Or is there anything I can do to fix this?

here's the configuration I'm using in launch.json

```json
{
    "name": ".NET Core Kubernetes Attach",
    "type": "coreclr",
    "request": "attach",
    "processId": "${command:pickRemoteProcess}",
    "justMyCode": false,
    "pipeTransport": {
        "pipeProgram": "kubectl",
        "pipeArgs": ["exec", "-n", "test", "-i", "${command:extension.vsKubernetesSelectPod}", "--"],
        "pipeCwd": "${workspaceFolder}",
        "debuggerPath": "/bin/vsdbg/vsdbg",
        "quoteArgs": false,
    },
    "sourceFileMap": {
        "/src": "${workspaceFolder}"
    },
    "logging": {
        "diagnosticsLog": {
            "protocolMessages": true,
            "debugEngineAPITracing": "error",
            "debugRuntimeEventTracing": true,
            "dispatcherMessages": "error",
            "expressionEvaluationTracing": true,
            "startDebuggingTracing": true
        }
    }
}
```

**Comment by tommcdon** [comment]
Hi @arminbashizade! We are reviewing older bugs and came across this one.  We apologize for the delayed response!  Are you still experiencing issues with debugging and Kubernetes?  The error number 0x80131124 is [CLDB_E_INDEX_NOTFOUND](https://github.com/dotnet/runtime/blob/c40c261a4219dd47690e70319b495fa9b1c36d0e/src/coreclr/inc/corerror.xml#L247), which is a metadata error.  This *could* be related to dynamically injected code into the process.  Did you happen to have a profiling agent in the pod re-writing IL dynamically?  One way a profiler can be injected is with the [`CORECLR_ENABLE_PROFILING`](https://learn.microsoft.com/dotnet/core/runtime-config/debugging-profiling) environment variable.

**Comment by arminbashizade** [comment]
Hi @tommcdon, I haven't tried this in some time, let me try again and get back to you

**Comment by arminbashizade** [comment]
so I tried again and I don't have that issue anymore, not sure if you updated something or if I had changed something since then
